src/_pytest/cacheprovider.py

```diff
@@ -210,9 +210,7 @@ def pytest_runtest_logreport(self, report):
     def pytest_collectreport(self, report):
         passed = report.outcome in ("passed", "skipped")
         if passed:
-            if report.nodeid in self.lastfailed:
-                self.lastfailed.pop(report.nodeid)
-                self.lastfailed.update((item.nodeid, True) for item in report.result)
```

Comment:
This line is pretty weird... wasn't it just adding the `nodeids` back to `lastfailed`? 🤔 

`git blame` says @RonnyPfannschmidt was the previous author, do you remember what's this line suppose to do @RonnyPfannschmidt?

Reply:
> git blame

Do you have a commit?

The other day when I've looked at this I've noticed that this is for collection: if collection failed before it gets popped and then all tests from the collection report are considered to be failures (so they are re-run I assume).
So it might make sense to have in general, but should have a test.

Reply:
Thanks for the clarification 👍